tron/utils/scribereader.py

```diff
@@ -90,146 +141,108 @@ def read_log_stream_for_action_run(
-                ):
-                    output.append((payload["timestamp"], payload["message"]))
-
-    if use_tailer:
-        stream = scribereader.get_stream_tailer(
-            stream_name=stream_name,
-            tailing_host=host,
-            tailing_port=port,
-            lines=-1,
-        )
-        try:
-            for line in stream:
-                if num_lines == max_lines:
-                    truncated_output = True
-                    break
-                # it's possible for jobs to run multiple times a day and have obscenely large amounts of output
-                # so we can't just truncate after seeing X number of lines for the run number in question - we
-                # need to count how many total lines we've seen and bail out early to preserve tron's uptime
-                num_lines += 1
-
-                try:
-                    payload = json.loads(line)
-                except json.decoder.JSONDecodeError:
-                    log.error(f"Unable to decode log line from stream ({stream_name}) for {action_run_id}: {line}")
-                    malformed_lines += 1
-                    continue
-
-                if (
-                    payload.get("tron_run_number") == int(run_num)
-                    and payload.get("component") == component
-                    and payload.get("message") is not None
-                    and payload.get("timestamp") is not None
-                    and payload.get("cluster") == paasta_cluster
-                ):
-                    output.append((payload["timestamp"], payload["message"]))
-        except StreamTailerSetupError:
-            return [
-                f"No data in stream {stream_name} - if this is the first time this action has run and you expected "
-                "output, please wait a couple minutes and refresh."
-            ]
-        except socket.timeout:
-            return [
-                f"Unable to connect to stream {stream_name} - if this is the first time this action has run and you "
-                "expected output, please wait a couple minutes and refresh."
-            ]
-        finally:
-            stream.close()
-
-    # XXX: for some reason, we're occasionally getting data out of order from scribereader - so we'll sort based on
-    # timestamp until we can figure out what's causing this.
-    output.sort(key=operator.itemgetter(0))
-    lines = [line for _, line in output]
-    malformed = [f"{malformed_lines} encountered while retrieving logs"] if malformed_lines else []
+
+    # yelp_clog S3LogsReader is a newer reader that is supposed to replace scribe readers eventually.
+    if use_s3_reader:
+        end_date = max_date.date() if max_date else today
+
+        log.debug("Using S3LogsReader to retrieve logs")
+        s3_reader = S3LogsReader(ecosystem).get_log_reader(log_name=stream_name, min_date=start_date, max_date=end_date)
+        paasta_logs.iterate_logs(s3_reader, max_lines)
```

Comment:
i wonder if we should rename `iterate_logs` as we're really fetching them - not returning an iterator

Reply:
also: thoughts on having this function return the log line? atm, it reads a bit weirdly since you'd expect a return value - but we're actually having the class store the log lines instead

Reply:
agree, it's not ideal but the primary reason for using class store is to avoid repeated sorting of lines when iterator is called multiple times for different readers, as in case of scribereader reader & tailler. With class store we  can call sort once after all the logs are fetched. Even though with the current `max_lines` pretty low this is not an issue at all, I was wondering if we should increase it to reduce not so rare occurrences of `This output is truncated` for certain jobs.

Reply:
increasing max_lines should be fine - it's more of a protection against jobs that spew gigabytes of logs and run multiple times :)

i'm hoping that at some point in the future we try to re-architect things a bit here so that we can store these logs in a more efficient way for tron to query them, but that's for future us!

Reply:
we're also currently only sorting once in the pre-PR code though, right?

Reply:
right, nothing changes in this regard. With `PaastaLogs` just try to avoid extra code duplication with addition of the the third reader. I think we can simplify it once again once we remove scribe readers

Reply:
so to rephrase:
- one point for using class store is to sort only once when all the logs have been fetched 
- secondly, to mask the details of paasta logs sorting inside the class itself (as we should care only about `lines`), rather than returning `output` with a list of tuples and doing sorting and extracting of `lines` outside it